src/tasks/base-chat-task.ts

```diff
@@ -0,0 +1,56 @@
+import { FastifyLoggerInstance } from 'fastify';
+import {
+  Actor,
+  DatabaseTransactionHandler,
+  IndividualResultType,
+  ItemMembershipService,
+  ItemService,
+  Member,
+  PostHookHandlerType,
+  PreHookHandlerType,
+  Task,
+  TaskStatus,
+} from 'graasp';
+import { ChatService } from '../db-service';
+
+export abstract class BaseChatTask<R> implements Task<Actor, R> {
+  protected itemService: ItemService;
+  protected itemMembershipService: ItemMembershipService;
+  protected chatService: ChatService;
+  protected _result: R;
+  protected _message: string;
```

Comment:
Why are we using the `_` for these props?

Reply:
Those are internal state mechanics between the parent and child task definitions that shouldn't be used for other purposes. I'm using the core code from @ahgentil so currently this is consistent with the rest of the server and plugins.